pinc/gradual.inc

```diff
@@ -127,4 +127,70 @@ function maybe_output_new_proofer_message()
+{
+    global $code_url, $pguser;
+
+    if($round != 'P1')
+        return;
+
+    // see if the user already has access to P3 or P2, if they do just
+    // encourage them to proof there instead
+    $userSettings =& Settings::get_Settings($pguser);
+    $p2_access = $userSettings->get_boolean("P2.access");
+    $p3_access = $userSettings->get_boolean("P3.access");
+    if($p2_access || $p3_access)
+    {
+        if($p3_access)
+            $round_target = "P3";
+        else
+            $round_target = "P2";
+
+        echo "<div class='callout'>";
+        echo "<div class='calloutheader'>";
+        echo _("Consider working on projects in $round_target!");
+        echo "</div>";
+
+        echo "<p>Please consider proofreading a project in $round_target rather than here in P1. With the large influx of new users, having eligible users working in <a href='$code_url/tools/proofers/round.php?round_id=$round_target'>$round_target</a> helps significantly.</p>";
+        echo "<p>Thank you!</p>";
+        echo "</div>";
+        return;
+    }
+
+    // if they are already eligible but have not requested access, tell them
+    $round = get_Round_for_round_id("P2");
+    $uao = $round->user_access($pguser);
+    if($uao->all_minima_satisfied)
+    {
+        echo "<div class='callout'>";
+        echo "<div class='calloutheader'>";
+        echo _("You are eligible to work in P2!");
+        echo "</div>";
+
+        echo "<p>You are eligible to work on projects in P2! Simply <a href='$code_url/tools/proofers/round.php?round_id=P2#Entrance_Requirements'>request access</a> and find a <a href='$code_url/tools/proofers/round.php?round_id=P2'>P2</a> project that interests you.</p>";
+        echo "<p>Please consider working on P2 projects rather than projects here in P1. With the large influx of new users, having eligible users working in <a href='$code_url/tools/proofers/round.php?round_id=P2'>P2</a> helps significantly.</p>";
+        echo "<p>Thank you!</p>";
+        echo "</div>";
+        return;
+    }
+
+    // they aren't yet eligible, but if they've been here for the required
+    // number of days and done the required number of pages and just need to
+    // take and pass the quiz, so nudge them along
+    $uao = $round->user_access($pguser);
+    if($uao->minima_table["P1"][3] && $uao->minima_table["days since reg"][3])
+    {
+        echo "<div class='callout'>";
+        echo "<div class='calloutheader'>";
+        echo _("You are a quiz away from working in P2!");
+        echo "</div>";
+
+        echo "<p>You are almost eligible to work in P2! The only thing left is to take and pass the <a href='$code_url/quiz/start.php?show_level=P_MOD'>moderate proofreading quiz</a>. After you pass the quiz you can <a href='$code_url/tools/proofers/round.php?round_id=P2#Entrance_Requirements'>request access</a> and you'll be able to work on projects in <a href='$code_url/tools/proofers/round.php?round_id=P2'>P2</a>.</p>";
```

Comment:
Shouldn't that be `quizzes`? Also, are all 10 moderate quizzes required, or just the first group of 5?

Reply:
How about "just a few short quizzes away"? Also could we say "working in the P2 round"? These are beginners and may not recognize what P2 means. The second use could be just "P2" though.

Reply:
Also -- "If you're qualified to work in a higher round, please consider working in that round." might be better for the first sentence "please consider...." since we would like P3s and F2s especially to work in those round.

Reply:
I'll get the first changes in.

Proofreaders who are already able to work in P2 or P3 will get the first message in this change so we don't need to tweak this message to apply to them.

Reply:
I've adjusted the wording. I'm trying to keep the first bold line short and succinct.

Reply:
Looks good to me.